src/detours.cpp

```diff
@@ -24,6 +25,78 @@ DECLARE_DETOUR(UTIL_SayText2Filter, Detour_UTIL_SayText2Filter, &modules::server
 DECLARE_DETOUR(IsHearingClient, Detour_IsHearingClient, &modules::engine);
 DECLARE_DETOUR(CSoundEmitterSystem_EmitSound, Detour_CSoundEmitterSystem_EmitSound, &modules::server);
 DECLARE_DETOUR(CCSWeaponBase_Spawn, Detour_CCSWeaponBase_Spawn, &modules::server);
+DECLARE_DETOUR(TriggerPush_Touch, Detour_TriggerPush_Touch, &modules::server);
+
+void FASTCALL Detour_TriggerPush_Touch(CTriggerPush* pPush, Z_CBaseEntity* pOther)
+{
+	MoveType_t movetype = pOther->m_MoveType();
+
+	// VPhysics handling doesn't need any changes
+	if (movetype == MOVETYPE_VPHYSICS)
+	{
+		TriggerPush_Touch(pPush, pOther);
+		return;
+	}
+
+	Z_CBaseEntity* pPushEnt = (Z_CBaseEntity*)pPush;
+
+	// SF_TRIG_PUSH_ONCE is handled fine already
+	if (pPushEnt->m_spawnflags() & 0x80)
+	{
+		TriggerPush_Touch(pPush, pOther);
+		return;
+	}
+
+	if (movetype == MOVETYPE_NONE || movetype == MOVETYPE_PUSH || movetype == MOVETYPE_NOCLIP)
+		return;
+
+	CCollisionProperty* collisionProp = pOther->m_pCollision();
+	if (!IsSolid(collisionProp->m_nSolidType(), collisionProp->m_usSolidFlags()))
+		return;
+
+	if (!pPush->PassesTriggerFilters(pOther))
+		return;
+
+	if (pOther->m_CBodyComponent()->m_pSceneNode()->m_pParent())
+		return;
+
+	Vector vecAbsDir;
+
+	matrix3x4_t mat = pPushEnt->m_CBodyComponent()->m_pSceneNode()->EntityToWorldTransform();
+	
+	Vector pushDir = pPush->m_vecPushDirEntitySpace();
+
+	// i had issues with vectorrotate on linux so i did it here
+	vecAbsDir.x = pushDir.x * mat[0][0] + pushDir.y * mat[0][1] + pushDir.z * mat[0][2];
+	vecAbsDir.y = pushDir.x * mat[1][0] + pushDir.y * mat[1][1] + pushDir.z * mat[1][2];
+	vecAbsDir.z = pushDir.x * mat[2][0] + pushDir.y * mat[2][1] + pushDir.z * mat[2][2];
+
+	Vector vecPush = vecAbsDir * pPush->m_flPushSpeed();
+
+	uint32 flags = pOther->m_fFlags();
+
+	if (flags & (0x800000))
+	{
+		vecPush = vecPush + pOther->m_vecBaseVelocity();
+	}
+
+	// if push up and FL_ONGROUND
+	if (vecPush.z > 0 && (flags & 1))
+	{
+		addresses::SetGroundEntity(pOther, nullptr);
+		Vector origin = pOther->GetAbsOrigin();
+		origin.z += 1.0f;
+
+		//CBaseEntity::Teleport
+		CALL_VIRTUAL(void, 148, pOther, origin, nullptr, nullptr);
+	}
+
+	pOther->m_vecBaseVelocity(vecPush);
+
+	// Add FL_BASEVELOCITY flag
+	flags |= (0x800000);
```

Comment:
I'm guessing the flags defined in the sdk are different to this? Because otherwise you can just put FL_BASEVELOCITY in there.